extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClientRecorder.java

```diff
@@ -10,44 +11,44 @@
 import com.mongodb.event.ConnectionPoolListener;
 
 import io.quarkus.arc.Arc;
-import io.quarkus.arc.runtime.BeanContainer;
-import io.quarkus.arc.runtime.BeanContainerListener;
 import io.quarkus.mongodb.reactive.ReactiveMongoClient;
 import io.quarkus.runtime.RuntimeValue;
 import io.quarkus.runtime.annotations.Recorder;
 
 @Recorder
 public class MongoClientRecorder {
 
-    public static final String DEFAULT_MONGOCLIENT_NAME = "<default>";
-    public static final String REACTIVE_CLIENT_NAME_SUFFIX = "reactive";
+    public Supplier<MongoClientSupport> mongoClientSupportSupplier(List<String> codecProviders, List<String> bsonDiscriminators,
+            List<ConnectionPoolListener> connectionPoolListeners, boolean disableSslSupport) {
+        return new Supplier<MongoClientSupport>() {
+            @Override
+            public MongoClientSupport get() {
+                return new MongoClientSupport(codecProviders, bsonDiscriminators, connectionPoolListeners, disableSslSupport);
+            }
+        };
+    }
 
-    public BeanContainerListener addMongoClient(
-            Class<? extends AbstractMongoClientProducer> mongoClientProducerClass,
-            boolean disableSslSupport) {
-        return new BeanContainerListener() {
+    public Supplier<MongoClient> mongoClientSupplier(String clientName,
+            @SuppressWarnings("usused") MongodbConfig mongodbConfig) {
```

Comment:
If it's unsed, why keeping it ?

Reply:
We need it to make sure that this recorder method is executed after the config system has been setup - see the comment in MongoProcessor where `mongoClientSupplier` is called

Reply:
OK, the fact that the comment is on the call site and not the parameter declaration site make me miss that point.